smart_proxy.go

```diff
@@ -148,16 +167,48 @@ func updateGroupInfo(servicesConf services.Configuration, groupsChannel chan []g
 		select {
 		case <-uptimeTicker.C:
 			groups, err = services.GetGroups(servicesConf)
-			if err != nil {
-				log.Error().Err(err).Msg("Error retrieving groups")
-			} else {
-				currentGroups = groups
-			}
+
+			currentGroups, currentErrorFound, currentError = handleGroupError(err, currentGroups, groups)
+
+			// empty the channels since they may contain old information
+			emptyChannels(groupsChannel, errorFoundChannel, errorChannel)
+		case errorFoundChannel <- currentErrorFound:
+		case errorChannel <- currentError:
 		case groupsChannel <- currentGroups:
 		}
 	}
 }
 
+// handleGroupError handles error after retrieving groups info in updateGroupInfo
+func handleGroupError(err error,
+	currentGroups []groups.Group,
+	newGroups []groups.Group) ([]groups.Group, bool, error) {
+	if err != nil {
+		log.Error().Err(err).Msg("Error retrieving groups")
+		var e *url.Error
+		if errors.As(err, &e) {
+			return currentGroups, true, &server.ContentServiceUnavailableError{}
+		}
+		return currentGroups, true, err
+	}
+	return newGroups, false, nil
+}
+
+// emptyChannels empties all channels related to updating group info
+func emptyChannels(groupsChannel chan []groups.Group,
+	errorFoundChannel chan bool,
+	errorChannel chan error) {
+	for len(errorFoundChannel) > 0 {
+		<-errorFoundChannel
+	}
+	for len(errorChannel) > 0 {
+		<-errorChannel
+	}
+	for len(groupsChannel) > 0 {
+		<-groupsChannel
```

Comment:
LGTM overall, just one question, I see you initialize `errorFoundChannel` and `errorChannel` with size=1, but `groupsChannel` isn't, right? I don't remember how the groupsChannel works completely, but is it enough to empty it like this?